### PR TITLE
feat: Add event-driven file watching for WATCH zones (Issue #52)

### LINKED ISSUE #52: Enhancement: Event-Driven File Watching for WATCH Zones
## Description

Add event-driven file watching to WATCH zones, enabling instant updates when files/directories change instead of polling on fixed intervals.

## Background

Currently WATCH zones poll every N seconds:
```bash
:zone watch LOG 80 20 5s "tail -20 /var/log/app.log"
                      ^^
                      5-second polling interval
```

**Problem:** 0-5 second delay before updates appear

**Solution:** Use inotify (Linux) or file watching to trigger instant updates

---

## Proposed Enhancement

**New syntax option - file watching instead of interval:**

```bash
# Event-driven: updates instantly when file changes
:zone watch LOG 80 20 watch:/var/log/app.log "tail -20 {file}"

# Still support interval-based (backward compatible)
:zone watch CPU 40 10 5s "uptime"
```

**Format:**
```
:zone watch NAME W H watch:PATH "COMMAND"
                     ^^^^^^^^^^^
                     Watch this file/directory for changes
```

---

## Benefits

**Applies to ALL WATCH zones, not just logs:**

**1. Instant log updates:**
```bash
:zone watch APP_LOG 80 20 watch:/var/log/app.log "tail -20 {file}"
:zone watch NGINX 80 15 watch:/var/log/nginx/access.log "tail -20 {file}"
```

**2. Instant git status after commits:**
```bash
:zone watch GIT 50 12 watch:.git "git status --short"
# Updates immediately after git add/commit/checkout
```

**3. Instant file listings after changes:**
```bash
:zone watch FILES 60 15 watch:. "ls -lt | head -15"
# Updates when files added/removed/modified
```

**4. Watch config files for changes:**
```bash
:zone watch CONFIG 60 10 watch:/etc/nginx/nginx.conf "cat {file}"
```

**5. More efficient (no continuous polling):**
- 0 delay vs 0-N second delay
- Less CPU (event-driven vs polling)
- Better UX

---

### Phase 1: Linux/WSL (inotify)

```python
import inotify.adapters

class FileWatcher:
    def __init__(self, path: str, callback):
        self.path = path
        self.callback = callback
        self._inotify = inotify.adapters.Inotify()
        self._inotify.add_watch(path)
        
    def start(self):
        for event in self._inotify.event_gen():
            if event and ('IN_MODIFY' in event[1] or 'IN_CREATE' in event[1]):
                self.callback()  # Trigger zone refresh
```

**Watch types:**
- File watch: Trigger on `IN_MODIFY`
- Directory watch: Trigger on `IN_CREATE`, `IN_DELETE`, `IN_MODIFY`
- Git watch: Watch `.git/` directory for changes

---

### Phase 2: Cross-Platform Fallback

**macOS:** Use `FSEvents` or polling fallback
**Windows:** Use `ReadDirectoryChangesW` or polling fallback

```python
# Fallback to polling if inotify unavailable
if not INOTIFY_AVAILABLE:
    # Poll file mtime every 0.5s
    while not stop_event.is_set():
        current_mtime = os.path.getmtime(path)
        if current_mtime > last_mtime:
            callback()
            last_mtime = current_mtime
        time.sleep(0.5)
```

---

### Option A: Special interval value

```bash
:zone watch LOG 80 20 watch "tail -20 /var/log/app.log"
                      ^^^^^
                      Keyword instead of interval
```

### Option B: Path parameter

```bash
:zone watch LOG 80 20 watch:/var/log/app.log "tail -20 {file}"
                      ^^^^^^^^^^^^^^^^^^^^^
                      Path to watch
```

**Recommendation:** Option B (more explicit, supports templates)

---

## Edge Cases

**1. File doesn't exist yet:**
- Wait for creation
- Show "Waiting for file..." message

**2. File deleted/rotated:**
- Detect and re-watch new file
- Handle gracefully (like `tail -f`)

**3. Directory watching:**
- Trigger on any file change in directory
- Useful for: `ls`, `git status`, etc.

**4. Permission errors:**
- Graceful degradation
- Show error message in zone

**5. Multiple zones watching same file:**
- Share single inotify watch
- Efficiency optimization

---

## Commands

```bash
# Create watch zone with file watching
:zone watch NAME W H watch:PATH "COMMAND"

# List all file watches
:zone watches

# Zone info shows watch type
:zone info LOG
# Output: "LOG (WATCH, file: /var/log/app.log, instant updates)"

# Existing pause/resume still work
:zone pause LOG
:zone resume LOG
```

---

## Implementation Checklist

- [ ] Add FileWatcher class
  - [ ] inotify support (Linux/WSL)
  - [ ] Directory watching
  - [ ] Multi-zone sharing (efficiency)
  - [ ] Stop/cleanup

- [ ] Integrate with ZoneExecutor
  - [ ] Parse `watch:PATH` syntax
  - [ ] Start file watcher thread
  - [ ] Trigger zone refresh on events
  - [ ] Handle errors gracefully

- [ ] Cross-platform support
  - [ ] macOS FSEvents (or polling fallback)
  - [ ] Windows fallback (polling)
  - [ ] Detect platform capabilities

- [ ] Edge cases
  - [ ] File doesn't exist (wait for creation)
  - [ ] File deleted/rotated
  - [ ] Permission errors
  - [ ] Multiple zones, same file

- [ ] Commands
  - [ ] Update `:zone watch` to accept watch:PATH
  - [ ] `:zone watches` - list active file watches
  - [ ] `:zone info` - show watch details

- [ ] Testing
  - [ ] Unit tests for FileWatcher
  - [ ] Integration tests with zones
  - [ ] Edge case handling
  - [ ] Cross-platform testing

- [ ] Documentation
  - [ ] Update CLAUDE.md
  - [ ] Add examples to layouts
  - [ ] Document in #51 (API guide)

---

## Success Criteria

- [ ] WATCH zones can watch files for instant updates
- [ ] WATCH zones can watch directories
- [ ] Git status updates instantly after commits
- [ ] Log files update instantly when appended
- [ ] Cross-platform support (Linux primary, fallback for others)
- [ ] No performance degradation
- [ ] Backward compatible (interval-based still works)

---

## Complexity

**Estimated effort:** 1-2 weeks

**Breakdown:**
- FileWatcher class: 3-4 days
- Integration with ZoneExecutor: 2-3 days
- Edge cases and testing: 2-3 days
- Cross-platform support: 2-3 days
- Documentation: 1 day

**Risk areas:**
- Cross-platform file watching (inotify Linux-only)
- Performance with many watched files
- Race conditions (file changes during read)

---

## Priority

**Medium-High** - Improves all WATCH zones significantly

**Benefits:**
- Better log monitoring (original #15 goal)
- Better git integration (supports #23 use cases)
- Better file monitoring
- More responsive zones overall

---

## Related Issues

- Closes #15 (Log Zone) - Better solution that benefits more use cases
- Enhances capabilities from #23 (Git Integration) - Instant git status
- Builds on existing WATCH zone infrastructure
- May benefit tmux integration (#19) - Watch tmux state changes

---

## Notes

**Alternative if too complex:**
- Close this issue
- Document WATCH zone patterns for logs in #51
- Use 1-2 second intervals for "good enough" responsiveness
- Revisit if users report polling delays as pain point

### PR DESCRIPTION
**Pull request by jcaldwell1066**
## Summary

Adds event-driven file watching for WATCH zones, enabling instant updates when files change instead of polling on fixed intervals.

**New syntax:**
```bash
# File-watching mode - instant updates when file changes
:zone watch LOG 80 20 watch:/var/log/app.log "tail -20 {file}"

# Interval mode - unchanged
:zone watch STATUS 80 20 5s "uptime"
```

## Changes

| File | Change |
|------|--------|
| `src/zones.py` | Add FileWatcher class, extend ZoneConfig with watch_path fields, integrate into ZoneExecutor |
| `src/main.py` | Update `:zone watch` command to detect `watch:PATH` syntax |
| `tests/test_zones.py` | Add 10 tests for FileWatcher and ZoneExecutor integration |
| `CLAUDE.md` | Document new syntax and examples |

## Features

- **inotify support** (Linux/WSL) - instant file change detection
- **Polling fallback** - cross-platform compatibility using mtime checks
- **Debouncing** - prevents rapid fire callbacks (0.5s default)
- **Template substitution** - `{file}` replaced with watched path in command
- **Edge case handling** - non-existent files, file creation, pause/resume
- **Backward compatible** - interval-based syntax unchanged

## Test plan

- [x] All 668 tests pass (10 new tests added)
- [x] FileWatcher polling detection works
- [x] Debouncing limits rapid callbacks
- [x] Template substitution works
- [x] Pause/resume works for file-watching zones
- [x] Documentation updated

Closes #52

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: src/zones.py [hunk 1]
```diff
@@ -1257,6 +1483,50 @@ def refresh_zone(self, name: str) -> bool:
...
+            zone.set_content(lines)
+            return result.returncode == 0
+
+        except subprocess.TimeoutExpired:
+            zone.set_content([f"[Command timed out after {timeout}s]"])
+            return False
+        except Exception as e:
+            zone.set_content([f"[Error: {e}]"])
+            return False
+
```
[the remark this answers is not in the document]
Comment (jcaldwell1066) [reply src/zones.py]: Fixed in 7feb9d8 - Refactored to delegate to `execute_pipe` via temporary command override as suggested. This eliminates the code duplication.

File: src/zones.py [hunk 2]
```diff
@@ -1183,11 +1214,205 @@ def from_dict(cls, data: dict) -> "ZoneManager":
...
+            # Set up watch flags
+            flags = inotify_simple.flags.MODIFY | inotify_simple.flags.CLOSE_WRITE
+            if is_directory:
+                flags |= (
+                    inotify_simple.flags.CREATE
+                    | inotify_simple.flags.DELETE
+                    | inotify_simple.flags.MOVED_TO
+                )
+
+            wd = self._inotify.add_watch(watch_path, flags)
```
[the remark this answers is not in the document]
Comment (jcaldwell1066) [reply src/zones.py]: Fixed in 7feb9d8 - Removed the unused `wd` variable.

File: src/zones.py [hunk 3]
```diff
@@ -1183,11 +1214,205 @@ def from_dict(cls, data: dict) -> "ZoneManager":
...
+
+    def _trigger_callback(self) -> None:
+        """Trigger callback with debouncing."""
+        now = time.monotonic()
+        if now - self._last_callback_time >= self._debounce:
+            self._last_callback_time = now
+            try:
+                self._callback()
+            except Exception:
+                pass  # Callback errors shouldn't stop the watcher
```
[the remark this answers is not in the document]
Comment (jcaldwell1066) [reply src/zones.py]: **Acknowledged - Deferred to logging enhancement.**

Same rationale as issue #4 - adding logging requires architectural decisions about where logs appear in a TUI application. Silent callback errors prevent the watcher thread from crashing, which is the primary goal.

In practice, the callback (`_on_file_change`) delegates to `execute_with_template` which already has its own error handling that displays errors in the zone content (e.g., "Error: ...").

Will address as part of a broader logging/observability improvement.

File: src/zones.py [hunk 4]
```diff
@@ -1183,11 +1214,205 @@ def from_dict(cls, data: dict) -> "ZoneManager":
...
+                    self._last_mtime = 0.0
+            except OSError:
+                pass  # File may be temporarily inaccessible
+
+            # Wait for next poll interval or stop signal
+            self._stop_event.wait(timeout=self._poll_interval)
+
+    def _watch_loop_inotify(self) -> None:
+        """Watch using inotify (Linux-only, more efficient)."""
+        import os
```
[the remark this answers is not in the document]
Comment (jcaldwell1066) [reply src/zones.py]: **Acknowledged - Minimal overhead, clarity benefit.**

The `os` module is already imported at file level in zones.py (line ~15). The local imports here are redundant but harmless - Python caches imported modules, so there's no performance impact.

These could be removed in a cleanup pass, but they don't affect functionality. Lower priority than the feature work.
